app/model/Challenge.php

```
@@ -126,15 +131,21 @@ public function updateChallenge($data)
 
-        if ($challengeData['created_by'] === $this->user->id || $this->user->getIdentity()->role <= Role::MODERATOR) {
+        if ($challengeData['created_by'] === $this->user->getIdentity()->id || $this->user->getIdentity()->role <= Role::MODERATOR) {
             $data['updated_at'] = $this->getDateTime();
 
             if (empty($data['start_at'])) {
-                $data['start_at'] = $this->getDateTime();
+                $data['start_at'] = $this->getDateTime('Y-m-d 00:00:00');
+            } else {
+                $d = DateTime::createFromFormat('Y/m/d', $data['start_at']);
+                $data['start_at'] = $d ? $d->format('Y-m-d 00:00:00') : $data['start_at'];
             }
```

PHPMD (2.15.0)

[error] The method updateChallenge uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)